### PR TITLE
Topic soho calendar

### PR DESCRIPTION
**Pull request by tlwalla1**
@jbest84 @CHenrie @jeremyphillips9998 Please review. This is the new month calendar view. I'll be bringing up concerns about the week view in the spring review.

### CODE REVIEW

File: src/Utility.js [hunk 1]
```diff
@@ -92,6 +92,18 @@ const __class = lang.setObject('crm.Utility', lang.mixin({}, Utility, {
     }
     return id;
   },
+  trimText: function trimText(text = {}, wordCount = {}) {
```
Comment (jbest84) [inline src/Utility.js]: I would default text and wordCount params to their correct value type. text = '' and wordCount = 0 as defaults.

File: src/Utility.js [hunk 2]
```diff
@@ -92,6 +92,18 @@ const __class = lang.setObject('crm.Utility', lang.mixin({}, Utility, {
...
+    const words = text.split(' ');
+    if (words.length > wordCount) {
+      const intermediate = words.slice(0, wordCount);
+      if (intermediate[wordCount - 1].slice(-1) === '.') {
```
Comment (jbest84) [inline src/Utility.js]: Babel polyfills es6 string methods, so instead of slice(-1) you can use `endsWith('.')`

File: src/Utility.js [hunk 3]
```diff
@@ -92,6 +92,18 @@ const __class = lang.setObject('crm.Utility', lang.mixin({}, Utility, {
...
+        intermediate[wordCount - 1] = intermediate[wordCount - 1].slice(0, -1);
+      }
+      const value = intermediate.join(' ') + ' ...';
```
Comment (jbest84) [inline src/Utility.js]: You can use string interpolation here: 
    `
    `${intermediate.join(' ')} ...`
`

File: src/ApplicationModule.js [hunk 4]
```diff
@@ -177,9 +175,7 @@ const __class = declare('crm.ApplicationModule', [ApplicationModule], {
       },
     }));
 
-    this.registerView(new CalendarMonthView());
```
Comment (CHenrie) [inline src/ApplicationModule.js:180]: What does this mean for existing customizations

File: src/Views/Calendar/CalendarView.js [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+    this.monthActivities = [];
+    this.activityQuery = this.formatQueryActivity(this.currentDate);
+    this.eventQuery = this.formatQueryEvent(this.currentDate);
+    const activityPromise = new Promise((resolve, reject) => { // eslint-disable-line
+      resolve(this.requestData());
+    });
+    const eventPromise = new Promise((resolve, reject) => { // eslint-disable-line
+      resolve(this.requestEventData());
+    });
+    Promise.all([activityPromise, eventPromise]).then(() => {
```
Comment (jbest84) [inline src/Views/Calendar/CalendarView.js]: `this.requestData()` and `this.requestEventData()` already return a deferred/promise. Promise.all shouldn't need the wrappers above it (activityPromise and eventPromise). It should be able to just do `Promise.all([this.requestData(), this.requestEventData()]).then(...)`

File: src/Utility.js (hunk 3 above)
Comment (CHenrie) [inline src/Utility.js]: Yes but with the space between ' ...'  make it harder read with the dollar sing and brackets, is there advantage

File: src/Views/Calendar/CalendarView.js (hunk 5 above)
Comment (CHenrie) [inline src/Views/Calendar/CalendarView.js]: good catch

File: src/Utility.js (hunk 3 above)
Comment (jbest84) [inline src/Utility.js]: They can do multi-line. They are easier to read as the string grows. I do not find it harder to read.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/template_strings

File: src/ApplicationModule.js (hunk 4 above)
Comment (jbest84) [inline src/ApplicationModule.js:180]: Keep the view registered. However, go into the view's properties and set expose to false. This way it won't show up in the configure menu anymore.

**Pull request merged**